resolver/src/lib.rs

```diff
@@ -38,12 +39,45 @@ pub trait Resolver: Clone + Send + 'static {
     /// Type used to uniquely identify data.
     type Key: Span;
 
+    /// Type used to identify peers for targeted fetches.
+    type PublicKey: PublicKey;
```

Comment:
My thinking here is that we'll just ignore stuff like this if we back the resolver with some API (still useful to know who told us) 👍